guides/login-with-github.mdx

```diff
@@ -0,0 +1,116 @@
+
+(**Note**: Using `ngrok` is just to get started with getting the GitHub OAuth2.0 flow working. In a production setting, if Flipt is publicly accessible, you can use that hostname instead of `ngrok`. These values can always be changed later after the creation of the application.)
+
+5. Retrieve the `Client ID` and `Client Secret` from the created OAuth2.0 app
+
+The Client ID should already be provided to you. You will have to generate a client secret. Click on the `Generate a new client secret` button (it might ask you to authenticate again with GitHub).
+
+## Running Flipt Locally
+
+1. Define a `config.yml`
+
+Flipt relies on configuration that the user provides for many bits of functionality. In order to enable the Login With GitHub feature, you must define a configuration file `config.yml` with certain fields and values.
+
+Our [documentation](https://www.flipt.io/docs/configuration/overview) gives a complete list of all configuration values available for how to configure Flipt.
+
+The [Authentication Methods: GitHub](/configuration/overview#authentication-methods-github) section of our configuration documentation describes the values needed to enable the Login with GitHub functionality.
+
+It should look similar to the following:
+
+```yaml
+version: "1.0"
+
+log:
+  level: DEBUG
+
+authentication:
+  required: true
+  session:
+    domain: ${REDIRECT_ADDRESS}
+  methods:
+    github:
+      enabled: true
+      client_id: ${CLIENT_ID}
+      client_secret: ${CLIENT_SECRET}
+      redirect_address: ${REDIRECT_ADDRESS}
+      scopes:
+        - "user:email"
+```
+
+The `client_id` and `client_secret` are going to be the values from your GitHub OAuth application. The `redirect_address` will be the `ngrok` URL with no path (e.g. https://36f1-2600-1700-1d64-daa0-444c-3f7e-ceb7-3825.ngrok-free.app). The `scopes` are entirely dependent on what level of access you would like the returned GitHub access token to have. The [GitHub documentation](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps) describes a list of valid scopes.
+
+The last bit of configuration is the session details. In order for the browser to establish a session to communicate with Flipt in an authenticated way, we must provide access details in an HTTP cookie whose value is a static token created by Flipt. This static token is created during the GitHub OAuth2.0 flow, and associated with the GitHub metadata retrieved from the GitHub API with the access token. The `domain` value will specify which hosts can receive a cookie. Since we only care about the host serving Flipt, you can provide the `ngrok` URL value but without the `https://` prefix (e.g. 36f1-2600-1700-1d64-daa0-444c-3f7e-ceb7-3825.ngrok-free.app).
```

Comment:
```suggestion
The last bit of configuration is the session details. In order for the browser to establish a session to communicate with Flipt in an authenticated way, we must provide access details in an HTTP cookie whose value is a static token created by Flipt. This static token is created during the GitHub OAuth2.0 flow, and associated with the GitHub metadata retrieved from the GitHub API with the access token. The `domain` value will specify which host can receive the cookie. Since we only care about the host serving Flipt, you can provide the `ngrok` URL value but without the `https://` prefix (e.g. 36f1-2600-1700-1d64-daa0-444c-3f7e-ceb7-3825.ngrok-free.app).
```